### PR TITLE
MM-8368, nginx proxy faq

### PR DESCRIPTION
**Pull request by AndersonWebStudio**
FAQ for NGINX proxy install document

### CODE REVIEW

File: source/install/config-proxy-nginx.rst [hunk 1]
```diff
@@ -85,3 +85,67 @@ NGINX is configured using a file in the ``/etc/nginx/sites-available`` directory
...
+    location ~ /api/v[0-9]+/(users/)?websocket$ {
+      proxy_pass            http://backend;
+      (...)
+      proxy_set_header      Host (your host name);
+      proxy_set_header      X-Forwarded-For $remote_addr;
+    }
+
+  You may need to add variants of the host name that clients may call. Your NGINX log will be helpful in diagnosing the problem.
+
+2. "How do I setup nginx-proxy with the mattermost-docker installation?"
```
Comment (lindy65) [inline source/install/config-proxy-nginx.rst]: Should "mattermost" be capitalised here? e.g. Mattermost?

File: source/install/config-proxy-nginx.rst [hunk 2]
```diff
@@ -85,3 +85,67 @@ NGINX is configured using a file in the ``/etc/nginx/sites-available`` directory
...
 Now that NGINX is installed and running, you can configure it to use SSL, which allows you to use HTTPS connections and the HTTP/2 protocol.
+
+NGINX FAQ
+=========
+
+1. "Why are Websocket connections returning a 403 error?"
+
+  This is likely due to a cross-origin check failing. A check is applied for websocket code to see if the Origin header is the same as the host header and it may not be.
+
+  .. code-block:: none
```
Comment (jasonblais) [inline source/install/config-proxy-nginx.rst]: What is the code block for?

File: source/install/config-proxy-nginx.rst [hunk 3]
```diff
@@ -85,3 +85,67 @@ NGINX is configured using a file in the ``/etc/nginx/sites-available`` directory
...
+
+    location ~ /api/v[0-9]+/(users/)?websocket$ {
+      proxy_pass            http://backend;
+      (...)
+      proxy_set_header      Host (your host name);
+      proxy_set_header      X-Forwarded-For $remote_addr;
+    }
+
+  You may need to add variants of the host name that clients may call. Your NGINX log will be helpful in diagnosing the problem.
```
Comment (jasonblais) [inline source/install/config-proxy-nginx.rst]: How can they add variants?

File: source/install/config-proxy-nginx.rst [hunk 4]
```diff
@@ -85,3 +85,67 @@ NGINX is configured using a file in the ``/etc/nginx/sites-available`` directory
...
+
+  If you are using SSL, you may also need to expose port 443. 
+
+3. "Why does NGINX fail when installing Gitlab CE with Mattermost on Azure?"
+
+  You may need to update the Callback URLs for the Application entry of Mattermost inside your Gitlab instance.
+
+  - Log into your GitLab instance as the admin
+  - Go to the Admin area -> Applications
+  - Then click Edit on GitLab-Mattermost
```
Comment (jasonblais) [inline source/install/config-proxy-nginx.rst]: Is it named "GitLab-Mattermost" for everyone?

File: source/install/config-proxy-nginx.rst (hunk 1 above)
Comment (AndersonWebStudio) [reply source/install/config-proxy-nginx.rst]: I leave this up to you guys, in my opinion I would leave it lower case to match the repo name that it references but I can change it.

Comment (jasonblais) [reply source/install/config-proxy-nginx.rst]: @AndersonWebStudio I see, that's a good point. Let's keep it as `mattermost-docker`.

Also, I hope that changes I committed are okay - we often do so directly in docs repo, but I think usually devs like to take care of changes themselves.

**Comment by AndersonWebStudio** [comment]
I changed the websocket FAQ, please review - after thinking about it some more I think the updated suggestion is better, and more in line with suggestions that developers have already made for the issue.

**Pull request merged**